src/utils.ts

```diff
@@ -574,7 +582,7 @@ export class MongoDBNamespace {
 
   static fromString(namespace?: string): MongoDBNamespace {
     if (!namespace) {
-      throw new MongoDriverError(`Cannot parse namespace from "${namespace}"`);
+      throw new MongoInvalidArgumentError(`Cannot parse namespace from "${namespace}"`);
```

Comment:
It seems like an error is only throw when namespace is the empty string. I'm in favor of changing the error message to reflect that, but I'll defer to the rest of the team since I don't know all of the context behind this test.

Reply:
It also throws when no argument is provided and I think the error message lets you know what went wrong pretty well

The message would look something like 

`MongoInvalidArgumentError: Cannot parse namespace from ""`

or 

`MongoInvalidArgumentError: Cannot parse namespace from "undefined"`

If you're still in favour to change it, I can go ahead, but that seems fairly clear to me

Reply:
Ahh good point, don't think needs a change then.

Reply:
is this function meant to be user facing? if not, the error should probably be runtime